File: go/fibonacci_sequence/fibonacci_sequence_9.go
// This program calculates the Fibonacci sequence up to a specified number of terms.
// The Fibonacci sequence is a series of numbers where each number is the sum of the two preceding ones, usually starting with 0 and 1.
// This program is designed to help programmers understand how to implement the Fibonacci sequence in Go.
// It includes detailed comments to explain each part of the code.

package main

import "fmt"

// Function to calculate the Fibonacci sequence up to n terms
func fibonacci(n int) []int {
    // Initialize a slice to hold the Fibonacci sequence
    fibSeq := make([]int, n)
    
    // The first term of the Fibonacci sequence is always 0
    fibSeq[0] = 0
    
    // The second term of the Fibonacci sequence is always 1
    if n > 1 {
        fibSeq[1] = 1
    }
    
    // Calculate the remaining terms of the Fibonacci sequence
    for i := 2; i < n; i++ {
        // Each term is the sum of the two preceding terms
        fibSeq[i] = fibSeq[i-1] + fibSeq[i-2]
    }
    
    return fibSeq
}

func main() {
    // Define the number of terms to calculate in the Fibonacci sequence
    var numTerms int
    
    // Prompt the user to enter the number of terms
    fmt.Print("Enter the number of terms: ")
    fmt.Scan(&numTerms)
    
    // Calculate the Fibonacci sequence
    fibSeq := fibonacci(numTerms)
    
    // Print the Fibonacci sequence
    fmt.Println("Fibonacci sequence:")
    for i, term := range fibSeq {
        fmt.Printf("Term %d: %d\n", i+1, term)
    }
}

File: vbnet/fibonacci_sequence/fibonacci_sequence_9.vb
' This program calculates the Fibonacci sequence up to a specified number of terms.
' The Fibonacci sequence is a series of numbers where each number is the sum of the two preceding ones.
' The sequence starts with 0 and 1, and each subsequent number is the sum of the previous two.
' This program is designed to help programmers understand the implementation of the Fibonacci sequence in Visual Basic .NET.

Module FibonacciSequence

    Sub Main()
        ' Declare variables to hold the number of terms and the current Fibonacci numbers
        Dim numTerms As Integer
        Dim firstNum As Integer = 0
        Dim secondNum As Integer = 1
        Dim nextNum As Integer

        ' Prompt the user to enter the number of terms for the Fibonacci sequence
        Console.Write("Enter the number of terms: ")
        numTerms = Convert.ToInt32(Console.ReadLine())

        ' Print the first two terms of the Fibonacci sequence
        Console.WriteLine("Fibonacci Sequence:")
        Console.Write(firstNum & " " & secondNum & " ")

        ' Loop to calculate and print the remaining terms of the Fibonacci sequence
        For i As Integer = 3 To numTerms
            ' Calculate the next term in the sequence
            nextNum = firstNum + secondNum

            ' Print the next term
            Console.Write(nextNum & " ")

            ' Update the previous two terms
            firstNum = secondNum
            secondNum = nextNum
        Next

        ' Wait for the user to press a key before closing the console window
        Console.WriteLine()
        Console.WriteLine("Press any key to exit.")
        Console.ReadKey()
    End Sub

End Module

